operator/api/v1alpha1/paladin_types.go

```diff
@@ -37,6 +37,10 @@ type PaladinSpec struct {
 	// (vs. configuring a connection to a production blockchain network)
 	BesuNode string `json:"besuNode,omitempty"`
 
+	// AuthConfig is used to provide authentication details for blockchain connections
+	// If this is set, it will override the auth details in the config
```

Comment:
I imagine we'll eventually need the same for Postgres auth security as well ?

Reply:
Agree with @onelapahead here that we should be explicit so that we leave room for future proofing.  That could be  a single`auth` struct with fields for the different clients as @onelapahead suggests.   An alternative spelling might be to extend (or wrap) the `besuNode` struct that we already have in the CRD.  Currently, it must be a string that is interpreted as a  reference to a `Besu` CR but maybe this could become a struct that could either contain a object reference or a set of creds (by reference to secret) and the URL to an actual blockchain node. e.g.

i.e. either of the following would be valid
```yaml
baseLedger:
  besuNode: objectRef
```
or

```yaml
baseLedger:
  endpoints:
    jsonRPCUrl: ...
    wsUrl: ...
    authConfig: 
      ...
```

Although, I am assuming here that the same auth would always be used for JSONRPC endpoint and WS endpoint.  Would need to confirm if that is a safe assumption.

Reply:
I was considering how much we want to "replicate" the Paladin configuration here. It's starting to resemble the Paladin config itself, so perhaps it would be better to define the Paladin config as a typed structure directly in the CRD definition, rather than keeping it as a blob